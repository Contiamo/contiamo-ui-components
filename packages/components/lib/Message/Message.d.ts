/// <reference types="react" />
import * as React from "react";
import { Css } from "../types";
export interface Props {
    css?: Css;
    className?: string;
    /** Message contents, can be any html element/React fragment. */
    children?: React.ReactNode;
    /** Background message color */
    color?: string;
    /** Called when close icon is clicked. Icon is not rendered at all if this prop is not specified. */
    onClose?: () => void;
}
declare const Message: (props: Props) => JSX.Element;
export default Message;
