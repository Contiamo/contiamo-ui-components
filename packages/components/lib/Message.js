"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var theme_1 = require("@operational/theme");
var utils_1 = require("@operational/utils");
var Icon_1 = require("./Icon");
var Container = glamorous_1.default.div(function (_a) {
    var theme = _a.theme, color = _a.color;
    var backgroundColor = theme_1.expandColor(theme, color) || theme.colors.info;
    var textColor = utils_1.readableTextColor(backgroundColor, [theme.colors.black, "white"]);
    return {
        backgroundColor: backgroundColor,
        overflow: "hidden",
        color: textColor,
        padding: theme.spacing * 1 / 2 + "px " + theme.spacing + "px",
        borderRadius: 4,
        position: "relative",
        maxWidth: 400,
    };
});
var Content = glamorous_1.default.div(function (_a) {
    var theme = _a.theme;
    return ({
        paddingRight: theme.spacing * 2.5
    });
});
var IconContainer = glamorous_1.default.div(function (_a) {
    var theme = _a.theme;
    return ({
        position: "absolute",
        top: 0,
        right: 0,
        cursor: "pointer",
        width: theme.spacing * 2.5,
        height: theme.spacing * 2.5,
        padding: theme.spacing * 0.5,
        "& svg": {
            width: "100%",
            height: "100%"
        },
        ":hover": {
            backgroundColor: "rgba(0, 0, 0, 0.1)"
        }
    });
});
var Message = function (props) { return (React.createElement(Container, { css: props.css, className: props.className, color: props.color },
    React.createElement(IconContainer, { onClick: props.onClose },
        React.createElement(Icon_1.default, { name: "X" })),
    React.createElement(Content, null, props.children))); };
exports.default = Message;
//# sourceMappingURL=Message.js.map