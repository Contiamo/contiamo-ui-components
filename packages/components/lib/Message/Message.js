"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var theme_1 = require("@operational/theme");
var utils_1 = require("@operational/utils");
var _1 = require("../");
var Container = glamorous_1.default.div(function (_a) {
    var theme = _a.theme, color = _a.color;
    var backgroundColor = theme_1.expandColor(theme, color) || theme.colors.info;
    var textColor = utils_1.readableTextColor(backgroundColor, [theme.colors.black, "white"]);
    return {
        backgroundColor: backgroundColor,
        color: textColor,
        overflow: "hidden",
        padding: theme.spacing * 1 / 2 + "px " + theme.spacing * 3.5 + "px " + theme.spacing / 2 + " " + theme.spacing,
        paddingRight: theme.spacing * 2.5,
        borderRadius: 4,
        minHeight: theme.spacing * 2.5,
        display: "flex",
        alignItems: "center",
        justifyContent: "space-between",
        position: "relative",
        maxWidth: 400,
    };
});
var IconContainer = glamorous_1.default.div(function (_a) {
    var theme = _a.theme;
    return ({
        position: "absolute",
        top: 0,
        right: 0,
        cursor: "pointer",
        width: theme.spacing * 2.5,
        height: theme.spacing * 2.5,
        padding: theme.spacing * 0.5,
        borderBottomLeftRadius: theme.borderRadius,
        borderTopRightRadius: theme.borderRadius,
        "& svg": {
            width: "100%",
            height: "100%",
        },
        ":hover": {
            backgroundColor: "rgba(0, 0, 0, 0.1)",
        },
    });
});
var Message = function (props) { return (React.createElement(Container, { css: props.css, className: props.className, color: props.color },
    React.createElement(IconContainer, { onClick: props.onClose },
        React.createElement(_1.Icon, { name: "X" })),
    props.children)); };
exports.default = Message;
//# sourceMappingURL=Message.js.map