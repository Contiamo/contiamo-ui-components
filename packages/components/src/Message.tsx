import * as React from "react"
import glamorous from "glamorous"
import { Theme, expandColor } from "@operational/theme"
import { readableTextColor, darken } from "@operational/utils"
import Icon from "./Icon"

export interface IProps {
  css?: {}
  className?: string
  children?: React.ReactNode
  color?: string
  onClose?: () => void
}

const Container = glamorous.div(({ theme, color }: { theme: Theme; color?: string }): {} => {
  const backgroundColor = expandColor(theme, color) || theme.colors.white
  const textColor = readableTextColor(backgroundColor, [theme.colors.black, "white"])

  return {
    backgroundColor,
    overflow: "hidden",
    color: textColor,
    padding: `${theme.spacing * 1 / 2}px ${theme.spacing}px`,
    borderRadius: 2,
    position: "relative",
    maxWidth: 400,
    border: `1px solid rgba(0, 0, 0, 0.2)`
  }
})

const Content = glamorous.div(({ theme }: { theme: Theme }): {} => ({
  paddingRight: theme.spacing * 2.5
}))

const IconContainer = glamorous.div(({ theme }: { theme: Theme }): {} => ({
  position: "absolute",
  top: 0,
  right: 0,
  cursor: "pointer",
  width: theme.spacing * 2.5,
  height: theme.spacing * 2.5,
  padding: theme.spacing * 0.5,
  "& svg": {
    width: "100%",
    height: "100%"
  },
  ":hover": {
    backgroundColor: "rgba(0, 0, 0, 0.1)"
  }
}))

export default (props: IProps) => (
  <Container css={props.css} className={props.className} color={props.color}>
    <IconContainer onClick={props.onClose}>
      <Icon name="X" />
    </IconContainer>
    <Content>{props.children}</Content>
  </Container>
)
