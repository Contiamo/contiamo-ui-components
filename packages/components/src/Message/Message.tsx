import * as React from "react"
import styled from "react-emotion"
import { OperationalStyleConstants, Theme, expandColor } from "@operational/theme"
import { readableTextColor, darken } from "@operational/utils"
import { Icon } from "../"
import { WithTheme, Css, CssStatic } from "../types"

export interface Props {
  className?: string
  /** Message contents, can be any html element/React fragment. */

  children?: React.ReactNode
  /** Background message color */

  color?: string
  /** Called when close icon is clicked. Icon is not rendered at all if this prop is not specified. */

  onClose?: () => void
}

const Container = styled("div")(
  ({
    theme,
    color,
  }: {
    theme?: OperationalStyleConstants & {
      deprecated: Theme
    }
    color?: string
  }): CssStatic => {
    const backgroundColor = expandColor(theme.deprecated, color) || theme.deprecated.colors.info
    const textColor = readableTextColor(backgroundColor, [theme.deprecated.colors.black, "white"])
    return {
      backgroundColor,
      color: textColor,
      overflow: "hidden",
      padding: `${(theme.deprecated.spacing * 1) / 2}px ${theme.deprecated.spacing * 3.5}px ${theme.deprecated.spacing /
        2}px ${theme.deprecated.spacing}px`,
      paddingRight: theme.deprecated.spacing * 2.5,
      // Icon space
      borderRadius: 4,
      minHeight: theme.deprecated.spacing * 2.5,
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      position: "relative",
      maxWidth: 400,
    }
  },
)

const IconContainer = styled("div")(
  ({ theme }: WithTheme): CssStatic => ({
    position: "absolute",
    top: 0,
    right: 0,
    cursor: "pointer",
    width: theme.deprecated.spacing * 2.5,
    height: theme.deprecated.spacing * 2.5,
    padding: theme.deprecated.spacing * 0.5,
    borderBottomLeftRadius: theme.deprecated.borderRadius,
    borderTopRightRadius: theme.deprecated.borderRadius,
    "& svg": {
      width: "100%",
      height: "100%",
    },
    ":hover": {
      backgroundColor: "rgba(0, 0, 0, 0.1)",
    },
  }),
)

const Message = (props: Props) => (
  <Container className={props.className} color={props.color}>
    <IconContainer onClick={props.onClose}>
      <Icon name="X" />
    </IconContainer>
    {props.children}
  </Container>
)

export default Message
