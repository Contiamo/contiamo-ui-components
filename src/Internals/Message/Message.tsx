import * as React from "react"
import tinycolor from "tinycolor2"
import { DefaultProps } from "../../types"
import styled from "../../utils/styled"

import { expandColor } from "../../utils/constants"
import { NoIcon } from "../../Icon/Icon"

export interface MessageProps extends DefaultProps {
  className?: string
  /** Message contents, can be any html element/React fragment. */
  children?: React.ReactNode
  /** Background message color */
  color?: string
  /** Called when close icon is clicked. Icon is not rendered at all if this prop is not specified. */
  onClose?: () => void
  /** Called when message is clicked. */
  onClick?: () => void
}

const Container = styled("div")<{ color_?: string }>(({ theme, color_ }) => {
  const backgroundColor = tinycolor(expandColor(theme, color_) || theme.color.primary)
    .setAlpha(0.9)
    .toString()
  return {
    backgroundColor,
    color: theme.color.white,
    overflow: "hidden",
    boxShadow: "0 2px 6px rgba(0, 0, 0, .15)",
    padding: "8px 52px 8px 16px",
    borderRadius: theme.borderRadius,
    minHeight: 36,
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    position: "relative",
    maxWidth: 400,
  }
})

const IconContainer = styled("div")(({ theme }) => ({
  position: "absolute",
  top: 0,
  right: 0,
  cursor: "pointer",
  width: 36,
  height: 36,
  borderBottomLeftRadius: theme.borderRadius,
  borderTopRightRadius: theme.borderRadius,
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  ":hover": {
    backgroundColor: "rgba(0, 0, 0, 0.1)",
  },
}))

export const Message: React.SFC<MessageProps> = ({ color, onClose, children, ...props }) => (
  <Container {...props} color_={color}>
    <IconContainer
      onClick={e => {
        e.stopPropagation()
        if (onClose) {
          onClose()
        }
      }}
    >
      <NoIcon />
    </IconContainer>
    {children}
  </Container>
)

export default Message
