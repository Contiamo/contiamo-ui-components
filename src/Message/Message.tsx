import * as React from "react"

import styled from "../utils/styled"

export interface MessageProps {
  title?: React.ReactNode
  body: React.ReactNode
  type: "warning" | "info" | "error"
}

const Container = styled("div")<{ type: MessageProps["type"] }>`
  padding: ${props => props.theme.space.content}px;
  margin-bottom: ${props => props.theme.space.big}px;

  ${props => {
    switch (props.type) {
      case "warning":
        return `
          background-color: rgba(255, 196, 0, 0.08);
          border: 1px solid rgba(186, 129, 0, 0.2);
        `
      case "info":
        return `
          background-color: rgba(20, 153, 206, 0.05);
          border: 1px solid rgba(20, 153, 206, 0.1);
        `
      case "error":
        return `
          background-color: rgba(154, 0, 0, 0.05);
          border: 1px solid rgba(154, 0, 0, 0.1);
        `
    }
  }} h3 {
    font-size: ${props => props.theme.font.size.small}px;
    margin: 0;
    margin-bottom: ${props => props.theme.space.small}px;
    color: ${props => (props.type === "error" ? props.theme.color.error : props.theme.color.text.default)};
  }

  p {
    font-size: ${props => props.theme.font.size.small}px;
    margin: 0;
    color: ${props => props.theme.color.text.default};
    word-break: break-word;
    line-height: 1.46;
  }
`

export const Message: React.SFC<MessageProps> = ({ title, body, ...props }) => (
  <Container {...props}>
    {title && <h3>{title}</h3>}
    <p>{body}</p>
  </Container>
)

export default Message
