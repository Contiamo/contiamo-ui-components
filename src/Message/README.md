A simple component to display a message

### Usage

```jsx
<>
  <Message type="info" body="This is an informative message" title="Info" />
  <Message type="warning" body="This is a warning message" title="Warning" />
  <Message type="error" body="This is an error message" title="Error" />
  <Message type="info" body="This is an informative message (without header)" />
  <Message type="warning" body="This is a warning message (without header)" />
  <Message type="error" body="This is an error message (without header)" />
</>
```
