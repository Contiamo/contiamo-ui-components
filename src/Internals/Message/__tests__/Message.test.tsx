import { render } from "enzyme"
import * as React from "react"
import wrapDefaultTheme from "../../../utils/wrap-default-theme"
import { Message as ThemelessMessage } from "../Message"

const Message = wrapDefaultTheme(ThemelessMessage)

describe("Message Component", () => {
  it("Should render", () => {
    const renderedComponent = render(<Message />)
    expect(renderedComponent).toMatchSnapshot()
  })
})
