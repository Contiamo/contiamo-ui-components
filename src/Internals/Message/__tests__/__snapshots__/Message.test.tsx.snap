// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component Should render 1`] = `
<div
  class="css-ftf61z"
>
  <div
    class="css-p29iif"
  />
  <div
    class="css-imevwx"
  >
    <div
      class="css-q9kd00"
    >
      <svg
        fill="currentColor"
        height="18"
        style="margin-left:0;margin-right:0;cursor:default;transition:fill .075s ease"
        viewBox="0 0 9.47 9.47"
        width="18"
      >
        <path
          d="M5.44 4.74L9.33.85a.5.5 0 0 0 0-.7.5.5 0 0 0-.71 0L4.74 4 .85.15a.48.48 0 0 0-.7 0 .48.48 0 0 0 0 .7L4 4.74.15 8.62a.5.5 0 0 0 .35.85.5.5 0 0 0 .35-.14l3.89-3.89 3.88 3.89a.52.52 0 0 0 .38.14.54.54 0 0 0 .36-.14.51.51 0 0 0 0-.71z"
        />
      </svg>
    </div>
  </div>
</div>
`;
